be1-go/channel/lao/mod_test.go

```diff
@@ -25,7 +26,114 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestBaseChannel_RollCallOrder(t *testing.T) {
+func TestLAOChannel_Subscribe(t *testing.T) {
+	keypair := generateKeyPair(t)
+
+	fakeHub, err := NewfakeHub(keypair.public, nolog, nil)
+	require.NoError(t, err)
+
+	m := message.Message{MessageID: "0"}
+
+	channel := NewChannel("channel0", fakeHub, m, nolog)
+
+	laoChannel, ok := channel.(*Channel)
+	require.True(t, ok)
+
+	relativePath := filepath.Join("..", "..", "..", "protocol",
+		"examples", "query", "subscribe")
+
+	file := filepath.Join(relativePath, "subscribe.json")
+	buf, err := os.ReadFile(file)
+	require.NoError(t, err)
+
+	var message method.Subscribe
+	err = json.Unmarshal(buf, &message)
+	require.NoError(t, err)
+
+	socket := &fakeSocket{id: "socket"}
+
+	channel.Subscribe(socket, message)
+
+	require.True(t, laoChannel.sockets.Delete("socket"))
```

Comment:
Wait, this means that we let clients subscribe to LAOs that do not exist?

Reply:
No, the hub does the check that the channel, we want to subscribe to, exists and if it exists then the LAO has been created. In the test, we bypass the hub by calling Subscribe directly on the channel. Maybe we can add another check in Subscribe to be sure that the channel, we want to subscribe to, corresponds to the channel that we actually subscribe to.

Reply:
I want to add that as the function is channel.Subscribe(), channel has to exist to be able to call the function